File: HumanEval/13.py
## BEGIN - PROMPT


def greatest_common_divisor(a: int, b: int) -> int:
    """ Return a greatest common divisor of two integers a and b
    >>> greatest_common_divisor(3, 5)
    1
    >>> greatest_common_divisor(25, 15)
    5
    """

## BEGIN - CANONICAL SOLUTION
    while b:
        a, b = b, a % b
    return a

## BEGIN - TEST


METADATA = {
    'author': 'jt',
    'dataset': 'test'
}


def check(candidate):
    assert candidate(3, 7) == 1
    assert candidate(10, 15) == 5
    assert candidate(49, 14) == 7
    assert candidate(144, 60) == 12

## BEGIN - CHECK
check(greatest_common_divisor)

File: HumanEvalJs/13.js
const assert = require('assert');

//// BEGIN - PROMPT

/**
 * Return a greatest common divisor of two integers a and b
 *
 * @param {number} a - First integer.
 * @param {number} b - Second integer.
 * @returns {number} Greatest common divisor.
 *
 * @example
 * greatestCommonDivisor(3, 5); // 1
 * greatestCommonDivisor(25, 15); // 5
 */
function greatestCommonDivisor(a, b) {
//// BEGIN - CANONICAL SOLUTION
    while (b) {
        [a, b] = [b, a % b];
    }
    return a;
}

//// BEGIN - TEST

const METADATA = {
    author: 'jt',
    dataset: 'test'
};

function check(candidate) {
    assert.strictEqual(candidate(3, 7), 1);
    assert.strictEqual(candidate(10, 15), 5);
    assert.strictEqual(candidate(49, 14), 7);
    assert.strictEqual(candidate(144, 60), 12);
}

//// BEGIN - CHECK
check(greatestCommonDivisor);